sql-plugin/src/main/scala/com/nvidia/spark/rapids/GpuParquetScan.scala

```diff
@@ -902,10 +944,19 @@ class MultiFileParquetPartitionReader(
     }
 
     closeOnExcept(table) { _ =>
-      if (!extraInfo.isCorrectedRebaseMode) {
-        (0 until table.getNumberOfColumns).foreach { i =>
-          if (RebaseHelper.isDateTimeRebaseNeededRead(table.getColumn(i))) {
-            throw RebaseHelper.newRebaseExceptionInRead("Parquet")
+      (0 until table.getNumberOfColumns).foreach { i =>
+        val col = table.getColumn(i)
+        // if col is a day
+        if (!extraInfo.isCorrectedRebaseMode && col.getType == DType.TIMESTAMP_DAYS) {
```

Comment:
Same comment about unnecessary type checks here and just below.  Would be nice to factor out this as a utility rather than copy+paste the code.